### DISCUSSION TITLE
WWSmag calculation

### DISCUSSION THREAD

**Discussion by giannoko** [fppimenta/rheoTool#75] [General]
Hello,

I am using rheoTool with of70. I have implemented a new constitutive model similar to ML-IKH model. I want to calculate the WSSmag by using the code found in the Aneurysm tutorial which is: 

```

PostProcessing
{
  functions
   (
     WSSmag
      {
        funcType           calcWSS;
        
        isTwoPhaseFlow               false;
        includePolymericStresses false; // Only in case of viscoelastic model

        enabled            true;
        evaluateInterval   1;       
      }
   );
}
```

For my case the  "includePolymericStresses" option is "true" because I have an EVP model.

When I am running the case I get this error : 

```
--> FOAM FATAL ERROR: 
Not implemented

From function virtual Foam::tmp<Foam::GeometricField<double, Foam::fvPatchField, Foam::volMesh> > Foam::constitutiveEq::etaSThermo() const in file constitutiveEqs/constitutiveEq/constitutiveEq.H at line 159.

FOAM aborting

#0  Foam::error::printStack(Foam::Ostream&) at ??:?
#1  Foam::error::abort() at ??:?
#2  Foam::constitutiveEq::etaSThermo() const at ??:?
#3  Foam::constitutiveEq::tauTotal() const at ??:?
#4  Foam::constitutiveModel::tauTotal() const at ??:?
#5  Foam::ppUtils::calcWSS::update() at ??:?
#6  Foam::ppUtilInterface::update() at ??:?

```

What is missing or what is wrong ? 

Thank you

**Comment by fppimenta** [comment]
The issue is the same as here https://github.com/fppimenta/rheoTool/discussions/71#discussion-3670229 and you can fix it in the same way while the new release is not out. However, you can also easily fix it by redefining etaSThermo (just make it equal to etaS) in your new model, which takes far less time.

**Reply by giannoko** [reply]
Thank you for your reply. I would like to know what do you mean by "redefining" etaSThermo? I defined a new parameter "etaSThermo" and made it equal to etaS but I still get this error.

**Comment by fppimenta** [comment]
That's C++ stuff. Something like that should work:

```
      virtual tmp<volScalarField> etaSThermo() const
      {         
        tmp<volScalarField> eta
        (
          new volScalarField
          (
            IOobject
            (
              "etaSThermo",
              U().instance(),
              U().db()
            ),
            U().mesh(),
            etaS(),
            extrapolatedCalculatedFvPatchField<scalar>::typeName
          )
        );
        return eta;
      }

**Reply by giannoko** [reply]
Thank you very much for your help, it works.
